False hint for preallocation

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version v1.23.6 built from (unknown, mod sum: "h1:dxnT1QFIpTeVoFUPaVDeFJJ+To++8ANYsQ2JIxJY02s=") on (unknown)
```

</details>
no configs
<details><summary>Config file</summary>

```console
$ cat .golangci.yml
no file
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.14 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/lol/.cache/go-build"
GOENV="/home/lol/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/lol/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build142025697=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint run -v --enable-all
INFO [config_reader] Config search paths: [./ /dev/shm /dev /]
INFO [lintersdb] Active 41 linters: [bodyclose deadcode depguard dogsled dupl errcheck funlen gochecknoglobals gochecknoinits gocognit goconst gocritic gocyclo godox gofmt goimports golint gomnd goprintffuncname gosec gosimple govet ineffassign interfacer lll maligned misspell nakedret prealloc rowserrcheck scopelint staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck whitespace wsl]
INFO [lintersdb] Active 41 linters: [bodyclose deadcode depguard dogsled dupl errcheck funlen gochecknoglobals gochecknoinits gocognit goconst gocritic gocyclo godox gofmt goimports golint gomnd goprintffuncname gosec gosimple govet ineffassign interfacer lll maligned misspell nakedret prealloc rowserrcheck scopelint staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck whitespace wsl]
INFO [loader] Go packages loading at mode 575 (deps|exports_file|files|imports|compiled_files|types_sizes|name) took 79.000942ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 227.914µs
INFO [runner/goanalysis_metalinter/goanalysis] analyzers took 0s with no stages
INFO [runner/unused/goanalysis] analyzers took 0s with no stages
INFO [runner] Processors filtering stat (out/in): diff: 1/1, max_same_issues: 1/1, identifier_marker: 1/1, filename_unadjuster: 1/1, path_prettifier: 1/1, skip_dirs: 1/1, max_from_linter: 1/1, path_shortener: 1/1, cgo: 1/1, skip_files: 1/1, uniq_by_line: 1/1, nolint: 1/1, max_per_file_from_linter: 1/1, source_code: 1/1, autogenerated_exclude: 1/1, exclude: 1/1, exclude-rules: 1/1
INFO [runner] processing took 223.729µs with stages: nolint: 62.805µs, exclude: 60.52µs, identifier_marker: 35.976µs, autogenerated_exclude: 17.415µs, source_code: 15.787µs, path_prettifier: 14.253µs, skip_dirs: 6.434µs, uniq_by_line: 2.908µs, max_same_issues: 1.875µs, cgo: 1.313µs, max_from_linter: 1.038µs, path_shortener: 995ns, max_per_file_from_linter: 765ns, filename_unadjuster: 728ns, diff: 405ns, exclude-rules: 303ns, skip_files: 209ns
INFO [runner] linters took 40.375115ms with stages: goanalysis_metalinter: 39.761763ms, unused: 322.148µs
main.go:15:2: Consider preallocating `a` (prealloc)
	var (
	^
INFO File cache stats: 1 entries of total size 288B
INFO Memory: 3 samples, avg is 68.9MB, max is 68.9MB
INFO Execution took 128.038057ms
```

</details>

## source code

``` golang
package main

import (
	"fmt"
	"math/rand"
)

const maxNew = 100

func other() []int {
	return make([]int, rand.Int63()%maxNew)
}

func main() {
	var (
		a []int
		b = []int{1, 2, 3, 4, 5}
	)

	for range b {
		a = append(a, other()...)
	}

	fmt.Printf("a: %d\nb: %d\n", len(a), len(b))
}
```

--- a/go.mod
+++ b/go.mod
@@ -6,6 +6,7 @@ require (
 	4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a
 	github.com/Djarvur/go-err113 v0.0.0-20200511133814-5174e21577d5
 	github.com/OpenPeeDeeP/depguard v1.0.1
+	github.com/alexkohler/prealloc v0.0.0-20210204145425-77a5b5dd9799
 	github.com/ashanbrown/forbidigo v1.1.0
 	github.com/ashanbrown/makezero v0.0.0-20201205152432-7b7cdbb3025a
 	github.com/bombsimon/wsl/v3 v3.1.0
@@ -26,7 +27,6 @@ require (
 	github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0
 	github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca
 	github.com/golangci/misspell v0.0.0-20180809174111-950f5d19e770
-	github.com/golangci/prealloc v0.0.0-20180630174525-215b22d4de21
 	github.com/golangci/revgrep v0.0.0-20180526074752-d9c87f5ffaf0
 	github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4
 	github.com/jgautheron/goconst v0.0.0-20201117150253-ccae5bf973f3

--- a/pkg/golinters/prealloc.go
+++ b/pkg/golinters/prealloc.go
@@ -4,7 +4,7 @@ import (
 	"fmt"
 	"sync"
 
-	"github.com/golangci/prealloc"
+	"github.com/alexkohler/prealloc/pkg"
 	"golang.org/x/tools/go/analysis"
 
 	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
@@ -32,7 +32,7 @@ func NewPrealloc() *goanalysis.Linter {
 
 		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
 			var res []goanalysis.Issue
-			hints := prealloc.Check(pass.Files, s.Simple, s.RangeLoops, s.ForLoops)
+			hints := pkg.Check(pass.Files, s.Simple, s.RangeLoops, s.ForLoops)
 			for _, hint := range hints {
 				res = append(res, goanalysis.NewIssue(&result.Issue{
 					Pos:        pass.Fset.Position(hint.Pos),

